File: Language/C#/Book/tmp/LINQ实战/Chapter04.CSharp/Chapter04.Win/FormBooks.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;

using LinqInAction.LinqBooks.Common;

namespace LinqInAction.Chapter04.Win
{
  public partial class FormBooks : Form
  {
    public FormBooks()
    {
      InitializeComponent();
    }

    private void FormBooks_Load(object sender, EventArgs e)
    {
      var query =
        from book in SampleData.Books
        where book.Title.Length > 10
        orderby book.Price
        select new { Book=book.Title, book.Price };

      dataGridView1.DataSource = query.ToList();
    }
  }
}

File: Language/C#/Book/tmp/LINQ实战/Chapter04.VB/Chapter04.Win/FormBooks.vb
Imports LinqInAction.LinqBooks.Common

Public Class FormBooks

  Private Sub FormBooks_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
    Dim query = _
      From book In SampleData.Books _
      Where book.Title.Length > 10 _
      Order By book.Price _
      Select New With {.Book = book.Title, book.Price}

    dataGridView1.DataSource = query.ToList()
  End Sub
End Class